src/main/java/com/officelife/actions/TakeItem.java

```diff
@@ -21,25 +21,31 @@ public TakeItem(State state, Class<T> itemClass) {
 
   @Override
   public boolean accept() {
-    // TODO there may be many items on this location?
 
     Optional<Coords> currentCoords = state.world.actorLocation(state.person.id());
 
     if (!currentCoords.isPresent()) {
       throw new RuntimeException("person " + state.person.id() + " is nowhere");
     }
 
-    Item item = state.world.items.get(state.world.itemLocations.get(currentCoords.get()));
-    if (!itemClass.isInstance(item)) {
-      throw new RuntimeException("item is not of class " + itemClass);
-    }
+    String itemId = state.world.itemsAtLocation(currentCoords.get())
+            .stream()
+            .filter(id -> {
+              Item itemAtLocation = state.world.items.get(id);
+
+              return itemClass.isInstance(itemAtLocation);
+            })
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException("Unable to find item of " + itemClass));
+
+    Item item = state.world.items.get(itemId);
 
     Coords coords = currentCoords.get();
 
-    if (!state.world.itemLocations.containsKey(coords) || !state.world.itemLocations.get(coords).equals(item.id())) {
+    if (!state.world.itemsAtLocation(coords).contains(item.id())) {
```

Comment:
I'm not sure this check is necessary after the one above, since to trip it, we must have done `world.items.put(<something other than item.id()>, item.id())`, which would affect things everywhere in a much worse way. Maybe we should abstract away item ids so we handle only items? (just `// TODO` for now)